src/Microsoft.AspNet.Razor.Test.Sources/CaseSensitiveTagHelperDescriptorComparer.cs

```diff
@@ -60,6 +65,14 @@ public override int GetHashCode(TagHelperDescriptor descriptor)
                 hashCodeCombiner.Add(requiredAttribute, StringComparer.Ordinal);
             }
 
+            if (descriptor.AllowedChildren != null)
+            {
+                foreach (var child in descriptor.AllowedChildren)
```

Comment:
Just realized this and other sorted collections should be sorted when adding to the hash code. Otherwise end up with equal descriptors that have different hash codes. Either fix that for this descriptor and file a follow-up bug for other affected `GetHashCode()` implementations or leave as-is in this PR and file the follow-up for the whole deal.